dispatch/tests/test_widget_fields.py

```diff
@@ -691,3 +694,77 @@ def test_event_to_json_no_data(self):
         data = None
 
         self.assertEqual(testfield.to_json(data), None)
+
+    def test_datetimefield(self):
+        testfield = DateTimeField('Test')
+
+        date = datetime.today()
+        date_sz = date.isoformat()
+
+        testfield.validate(date_sz)
+
+        self.assertEqual(testfield.prepare_data(date_sz), date)
+
+    def test_datetimefield_invalid_data(self):
+        testfield = DateTimeField('Test')
+        date_sz = 'garbage'
+
+        try:
+            testfield.validate(date_sz)
+            self.fail('Field date is invalid, exception should have been thrown')
+        except InvalidField:
+            pass
```

Comment:
there is also built-in support in django.TestCase via unittest for expecting an exception:
```py
with self.assertRaises(InvalidField):
    testfield.validate(date_sz)
```